### ISSUE TITLE
Nuitka clears files in build/ which cause issues with `versioneer.py`

### ISSUE THREAD

**Issue by pyrito** [Nuitka/Nuitka#2207]
## Version
```
1.5.6
Commercial: None
Python: 3.8.16 (default, Mar  1 2023, 21:19:10) 
Flavor: Anaconda Python
Executable: /Users/kvelayutham/opt/anaconda3/envs/py38/bin/python
OS: Darwin
Arch: x86_64
Version C compiler: /usr/bin/clang (clang).
```

## Issue Description

I'm currently using Nuitka as part of a build for a project as such:
```python
setup(
    name="test",
    command_options={
        "nuitka": {
            "--module": None,
            "--include-package": [
                "test",
            ],
            "--no-pyi-file": None,
        }
    },
    version=versioneer.get_version(),
    packages=find_packages(
        exclude=[...]
    ),
    install_requires=[... ],
    python_requires=">=3.8",
    license_files=("LICENSE.txt",),
    license="Other/Proprietary License",
    cmdclass=cmdclass,
    build_with_nuitka=True,
)
```

I build it as such: `python setup.py bdist_nuitka clean --all`

I currently use [versioneer](https://github.com/python-versioneer/python-versioneer) to keep track of my package versions, but noticed that my installed package has an invalid version of `0+unknown` (the `__version__` attribute). After digging into this more, I see that versioneer changes `_version.py` when files are getting copied over to `build/` during the wheel generation. After digging into the Nuitka source code, I can see that Nuitka clears all the python files in `build/` and uses the source files instead for the wheel build instead, which is a little tricky for me. Is there a workaround for this? I can't seem to find a clear flag/option that handles this case. I'm able to work around this for now by modifying the source code during the wheel build myself, but would prefer to not do so.

**Comment by kayhayen** [comment]
People are actively asking to call `clean` before building, which does that deletion too, does it not?

However, it sounds like Nuitka is compiling the wrong files, that would be interesting to fix. Check `nuitka\distutils\DistutilCommands.py` and maybe you can enhance it. It calls `addMainScriptDirectory` as a path to search, but otherwise, it will expect `sys.path` to provide the source, and for all I know it does. But quite possibly, we might have to only add the build folder, and might be able to find everything there? That would be great. Not sure about other modules, you e.g. choose to include, but it being the main script path, and not deleting the files until after compilation, would probably go ways.

**Comment by pyrito** [comment]
@kayhayen you are right about what `clean` does, but I noticed this behavior during intermediate stages of the Nuitka build before the entire `build` directory gets cleaned out. 

I read through the DistutilCommands code earlier (I needed to in order to find a work around), and noticed that the build files are getting cleared here: https://github.com/Nuitka/Nuitka/blob/develop/nuitka/distutils/DistutilCommands.py#L244-L250. What was the impetus for this decision originally?

**Comment by kayhayen** [comment]
Well, Nuitka replaces said copied code with a compiled version of it. So it feels like when it searches for `your_package` the file it finds, is the one to compile. That it does by locating your source folder from the setuptools config. However, in a sense, that may be wrong. Instead, it should take the job descriptions from the config (it does) and then find the code in the build folder, with that allowing setuptools in other stages, to modify it.

So, I feel a little bit of reorganisation when the deletion happens, and where the source files are located from, not the source folder then, but build folder apparently, it will work with these kinds of plugins to setuptools and automatically be better. If you can run the test suite of `tests\distutils` and maybe add your case after a modification, that would be good and probably not very hard. It's just that I do not have the time and patience now.

**Comment by pyrito** [comment]
@kayhayen I think I can give this a go. I'll try to open a PR within the next few days.

**Comment by kayhayen** [comment]
You will be very welcome. Thanks for helping here.

**Comment by pyrito** [comment]
@kayhayen I opened a PR here: https://github.com/Nuitka/Nuitka/pull/2223. Would appreciate any feedback!

**Comment by kayhayen** [comment]
I merged it already, it will be in 1.6, thanks for helping with this one.

**Comment by kayhayen** [comment]
This is part of the stable release 1.6 that I just made.
